packages/rsbuild-plugin-angular/src/lib/plugin/compilation/setup-with-paralell-compilation.ts

```diff
@@ -0,0 +1,44 @@
+import type { RsbuildConfig } from '@rsbuild/core';
+import { compileString as sassCompileString } from 'sass-embedded';
+import type { PluginAngularOptions } from '../../models/plugin-options';
+import { ParallelCompilation } from '@angular/build/src/tools/angular/compilation/parallel-compilation';
+import { transformFileSync } from '@swc/core';
+import { setupCompilation } from './setup-compilation';
+
+export async function setupCompilationWithParallelCompilation(
+  config: Pick<RsbuildConfig, 'source'>,
+  options: Pick<
+    PluginAngularOptions,
+    'jit' | 'inlineStylesExtension' | 'fileReplacements' | 'tsconfigPath'
+  >
+) {
+  const { rootNames, compilerOptions } = setupCompilation(config, options);
+  const parallelCompilation = new ParallelCompilation(options.jit ?? false);
+  const fileReplacements: Record<string, string> =
+    options.fileReplacements.reduce((r, f) => {
+      r[f.replace] = f.with;
+      return r;
+    }, {});
+
+  try {
+    await parallelCompilation.initialize(
+      config.source?.tsconfigPath ?? options.tsconfigPath,
+      {
+        ...compilerOptions,
+        fileReplacements,
+        modifiedFiles: new Set(rootNames),
+        async transformStylesheet(data) {
+          const result = sassCompileString(data);
+          return result.css;
+        },
```

Comment:
Can we use the extracted method?
https://github.com/Coly010/ng-rspack-build/pull/57/files#diff-19385d4481f41247716d1f1623c664ffecbff6554146096732e1be9f20b37adeR65

Also as a follow up - we should potentially explore `compileStringAsync()`
<img width="658" alt="image" src="https://github.com/user-attachments/assets/23a9fe84-d205-438f-87b0-f4d82803400d" />

Reply:
Nice catch! Created issue and marked the code with a todo.